DataGrid/Extension/Configuration/EventSubscriber/ConfigurationBuilder.php

```diff
@@ -38,49 +48,78 @@ public static function getSubscribedEvents(): array
     public function readConfiguration(DataGridEventInterface $event): void
     {
         $dataGrid = $event->getDataGrid();
-        $dataGridConfiguration = [];
-        foreach ($this->kernel->getBundles() as $bundle) {
-            if ($this->hasDataGridConfiguration($bundle->getPath(), $dataGrid->getName())) {
-                $configuration = $this->getDataGridConfiguration($bundle->getPath(), $dataGrid->getName());
+        $mainConfiguration = $this->getMainConfiguration($dataGrid->getName());
+        if (null !== $mainConfiguration) {
+            $this->buildConfiguration($dataGrid, $mainConfiguration);
+        } else {
+            $this->buildConfigurationFromRegisteredBundles($dataGrid);
+        }
+    }
 
-                if (is_array($configuration)) {
-                    $dataGridConfiguration = $configuration;
-                }
-            }
+    private function buildConfiguration(DataGridInterface $dataGrid, array $configuration): void
+    {
+        foreach ($configuration['columns'] as $name => $column) {
+            $dataGrid->addColumn($name, $column['type'] ?? 'text', $column['options'] ?? []);
+        }
+    }
+
+    private function getMainConfiguration(string $dataGridName): ?array
+    {
+        $directory = $this->kernel->getContainer()->getParameter(self::MAIN_CONFIG_DIRECTORY);
+        if (null === $directory || false === is_dir($directory)) {
```

Comment:
if `$directory` does not point to a filesystem directory then an exception should be thrown